Sources/AblyAssetTrackingInternal/WorkerQueue/Worker.swift

```diff
@@ -0,0 +1,88 @@
+import Foundation
+import AblyAssetTrackingCore
+
+/// A [Worker] interface represents workers which execute work inside [WorkerQueue].
+/// PropertiesType: - the type of properties used by this worker as both input and output.
+/// WorkerSpecificationType - the type of specification used to post another worker back to the queue.
+public protocol Worker<PropertiesType, WorkerSpecificationType>: AnyObject {
+    associatedtype PropertiesType
+    associatedtype WorkerSpecificationType
+        
+    typealias WorkerAsyncWorkCompletionHandler = (Error?) -> Void
```

Comment:
Would be good to talk this through - in Android we use a `try/catch` for error handling so would be interested to know why we've done it differently here

Reply:
For clarity - it was changed as a result of this [discussion](https://github.com/ably/ably-asset-tracking-swift/pull/536#discussion_r1099167715)

But as you mentioned in a slack conversation - "We call the async work literally fire and forget". We don't really care about when the async work ends successfully, and if it fails - we can throw/catch an error. So we might not need a completion handler here at all, and the first version of this implementation might be correct after all.

Thoughts @lawrence-forooghian?

Reply:
We discussed this one on our call – I explained how an Android `suspend` function would be idiomatically represented in Swift as a function that takes a completion handler. Going to resolve this conversation on your behalf @AndyTWF.